packages/backend/src/open_payments/grantReference/service.test.ts

```diff
@@ -79,4 +80,60 @@ describe('Grant Reference Service', (): void => {
       )
     })
   })
+
+  describe('Get or Create Grant Reference', (): void => {
+    test('cannot fetch a non-existing grant reference', async (): Promise<void> => {
+      expect(
+        await grantReferenceService.getOrCreate(
+          { id: uuid(), clientId: uuid() },
+          AccessAction.List
+        )
+      ).toBeUndefined()
+    })
+
+    test('throws an error when clientId does not match', async (): Promise<void> => {
+      const id = uuid()
+
+      await grantReferenceService.create({
+        id,
+        clientId: uuid()
+      })
+
+      await expect(
+        async () =>
+          await grantReferenceService.getOrCreate(
+            { id, clientId: uuid() },
+            AccessAction.List
+          )
+      ).rejects.toThrowError('does not match internal reference clientId')
```

Comment:
I thought we would have needed a RegEx here, but it seems to work? 
```suggestion
      ).rejects.toThrowError(/does not match internal reference clientId/)
```

Reply:
I was thinking the same, but based on Jest docs, RegEx and strings are equivalent in this case.